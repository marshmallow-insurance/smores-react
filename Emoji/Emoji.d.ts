import { FC } from 'react';
import { MarginProps } from '../utils/space';
export type EmojiProps = {
    /** className attribute to apply classes from props */
    className?: string;
    /** Emoji symbol */
    symbol: string;
    /** set size of the Emoji  */
    size?: number;
    /** aria-label to identify Emoji */
    label: string;
} & MarginProps;
export declare const Emoji: FC<EmojiProps>;
