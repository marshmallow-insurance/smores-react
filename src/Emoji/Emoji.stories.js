import React from 'react'
import { Emoji } from './Emoji'

export default {
  title: 'Emoji',
  component: Emoji,
}

const Template = (args) => <Emoji {...args} />

export const Default = Template.bind({})

Default.args = {
  symbol: 'ðŸ˜‰',
  size: 32,
  label: 'wink',
  mt: '0',
  mr: '0',
  mb: '0',
  ml: '0',
}
