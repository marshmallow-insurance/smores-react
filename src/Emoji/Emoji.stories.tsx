import React from 'react'
import { Emoji, EmojiProps } from './Emoji'

export default {
  title: 'Emoji',
  component: Emoji,
}

const Template = (props: EmojiProps) => <Emoji {...props} />

export const Default = Template.bind({})

Default.args = {
  symbol: 'ðŸ˜‰',
  size: 32,
  label: 'wink',
  mt: '0',
  mr: '0',
  mb: '0',
  ml: '0',
}
