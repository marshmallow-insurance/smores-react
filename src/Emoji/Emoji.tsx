import React, { FC } from 'react'
import styled, { css } from 'styled-components'

type EmojiProps = {
  /** className attribute to apply classes from props */
  className?: string
  /** Emoji symbol */
  symbol: string
  /** set size of the Emoji  */
  size?: number
  /** aria-label to identify Emoji */
  label?: string
  /** margins TOP-RIGHT-BOTTOM-LEFT */
  mt?: string
  mr?: string
  mb?: string
  ml?: string
}

export const Emoji: FC<EmojiProps> = ({
  className = '',
  symbol,
  size = 12,
  label = 'emoji',
  mt = '0',
  mr = '0',
  mb = '0',
  ml = '0',
}) => (
  <Container
    className={className}
    size={size}
    mt={mt}
    mr={mr}
    mb={mb}
    ml={ml}
    role="img"
    aria-label={label}
  >
    {symbol}
  </Container>
)

interface IEmoji {
  size: number
  mt: string
  mr: string
  mb: string
  ml: string
}

const Container = styled.span<IEmoji>(
  ({ size, mt, mr, mb, ml }) => css`
    font-size: ${size}px;
    margin-top: ${mt};
    margin-right: ${mr};
    margin-bottom: ${mb};
    margin-left: ${ml};
  `,
)
