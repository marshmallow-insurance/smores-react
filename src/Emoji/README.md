# < Emoji />

## Props:

```ts
symbol:   string; /* copy/paste directly the emoji, not it's text representation. See example for reference */
size?:    number; /* emoji dimentions */
label:   string; /* aria-label */
```

## Usage

```js
import {Emoji} from '@mrshmllw/smores-react';

const App = () => (
  <Emoji symbol="ðŸ˜‰" size={24} label="wink"/>
);
```

## Emoji List:

You can find the full list [here](https://unicode.org/emoji/charts/full-emoji-list.html).
