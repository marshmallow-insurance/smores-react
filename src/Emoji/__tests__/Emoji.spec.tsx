import { render } from '@testing-library/react'
import React from 'react'

import { Emoji } from '../Emoji'

test('renders', () => {
  const { container } = render(<Emoji symbol="ðŸ˜‰" size={24} label="wink" />)
  expect(container.firstChild).toMatchSnapshot()
})
