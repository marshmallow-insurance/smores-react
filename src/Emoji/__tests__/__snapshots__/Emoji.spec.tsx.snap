// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<span
  aria-label="wink"
  class="sc-guDLey hzMujr"
  role="img"
>
  ğŸ˜‰
</span>
`;
